app/uk/gov/hmrc/gform/gformbackend/GformBackEndService.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.gformbackend
+
+import cats.instances.future._
+import play.api.mvc.Request
+import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
+import uk.gov.hmrc.gform.controllers.AuthCacheWithForm
+import uk.gov.hmrc.gform.gform.{ CustomerId, FrontEndSubmissionVariablesBuilder, StructuredFormDataBuilder }
+import uk.gov.hmrc.gform.graph.{ CustomerIdRecalculation, EmailParameterRecalculation, Recalculation }
+import uk.gov.hmrc.gform.lookup.LookupRegistry
+import uk.gov.hmrc.gform.sharedmodel.form.{ Form, FormId, FormStatus, UserData }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ EmailParametersRecalculated, FormTemplate }
+import uk.gov.hmrc.gform.sharedmodel.structuredform.StructuredFormValue
+import uk.gov.hmrc.gform.sharedmodel.{ AccessCode, AffinityGroupUtil, LangADT, SubmissionData }
+import uk.gov.hmrc.gform.submission.Submission
+import uk.gov.hmrc.gform.summary.{ SubmissionDetails, SummaryRenderingService }
+import uk.gov.hmrc.http.{ HeaderCarrier, HttpResponse }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+trait GformBackEndAlgebra[F[_]] {
+  def submissionDetails(formId: FormId)(implicit hc: HeaderCarrier): F[Submission]
+
+  def submitWithUpdatedFormStatus(
+    formStatus: FormStatus,
+    cache: AuthCacheWithForm,
+    maybeAccessCode: Option[AccessCode],
+    submissionDetails: Option[SubmissionDetails])(
+    implicit request: Request[_],
```

Comment:
Why is `Request[_]` needed? Isn't `HeaderCarrier` enough?

Reply:
Seems got be needed for SummaryRenderingService.createHtmlForPdf - I've tried removing it all the way down but hit a dead end.

Reply:
Ok